diff --git a/src/Wishlist/CoreBundle/Repository/PurchaseRepository.php b/src/Wishlist/CoreBundle/Repository/PurchaseRepository.php
index e87c137..f387ace 100644
--- a/src/Wishlist/CoreBundle/Repository/PurchaseRepository.php
+++ b/src/Wishlist/CoreBundle/Repository/PurchaseRepository.php
@@ -63,7 +63,7 @@ class PurchaseRepository extends EntityRepository
         
         // if user is purchasing for themselves, remove the existing purchase promise
         if(isset($selfPurchaseItem) && isset($purchasePromised))
-        {
+        {            
             $this->deletePurchase($purchasePromised, PurchaseEventTypes::RemovedFromWishlist);
         }
         
@@ -134,7 +134,7 @@ class PurchaseRepository extends EntityRepository
     public function deletePurchase($purchase, $event_type)
     {
         if(!isset($event_type) || !isset($purchase)){
-            throw new \Exception("An invalid purchase and/or event was passed in. Contact the Wishlist Admins for assistance.");
+            return false; // on_error
         }
         
         $em = $this->getEntityManager();
@@ -142,6 +142,8 @@ class PurchaseRepository extends EntityRepository
         $em->flush();
 
         $this->purchaseEventNotification($event_type);
+        
+        return true; // on_success
     }
 
     // TO DO: send notification to user. Explaining why the item was removed from 
diff --git a/src/Wishlist/CoreBundle/Repository/WishlistItemRepository.php b/src/Wishlist/CoreBundle/Repository/WishlistItemRepository.php
index decb8b8..53ebdef 100644
--- a/src/Wishlist/CoreBundle/Repository/WishlistItemRepository.php
+++ b/src/Wishlist/CoreBundle/Repository/WishlistItemRepository.php
@@ -100,8 +100,10 @@ class WishlistItemRepository extends EntityRepository
         }
         
         // search for and remove the promised purchase of this wish (if it exists)
-        $purchaseRepo = $this->getEntityManager()->getRepository('WishlistCoreBundle:Purchase');        
-        $purchaseRepo->deletePurchase($wishToDelete->getPurchase(), PurchaseEventTypes::RemovedFromWishlist);
+        $purchaseRepo = $this->getEntityManager()->getRepository('WishlistCoreBundle:Purchase');   
+        $thisPurchase = $wishToDelete->getPurchase();
+        $thisEventType = \Wishlist\CoreBundle\Entity\PurchaseEventTypes::RemovedFromWishlist;
+        $purchaseRepo->deletePurchase($thisPurchase, $thisEventType);
         
         $em->remove($wishToDelete);
         $em->flush();
diff --git a/web/js/wishlist.js b/web/js/wishlist.js
index 2b62161..0b5a5d7 100644
--- a/web/js/wishlist.js
+++ b/web/js/wishlist.js
@@ -130,12 +130,16 @@ function setupWishlist()
     })
 
     $("h3 .ui-icon-close").click(function(){
-        if(confirm("Are you sure you want to delete this wish?"))
-        {
-            $(wishlist_div).accordion( "option", "disabled", true);
-            var itemObj = {name: escape($(this).next().text())};
-            delFromWishlist(itemObj, setupWishlist);
-        }        
+        var itemToDelete = escape($(this).next().text());
+        confirm("Are you sure you want to delete this wish?")
+        .then(function (answer) {
+            if(answer == 1)
+            {
+                $(wishlist_div).accordion( "option", "disabled", true);
+                var itemObj = {name: itemToDelete};
+                delFromWishlist(itemObj, setupWishlist);
+            }
+        });        
     });
     
     $('.purchaseBtn').on('click', clickedItem);
